### DISCUSSION TITLE
BlenderBIM: Set the number of storeys when creating a new IFC Project

### DISCUSSION THREAD

**Discussion by brunoperdigao** [IfcOpenShell/IfcOpenShell#2301] [Ideas]
Hi, everyone,

I was trying to test this idea of creating a "number of storeys" parameter in the IFC Project menu:
![image](https://user-images.githubusercontent.com/57102715/180104357-8d534632-0765-473b-bf19-d8dfb5477d3a.png)

First, I'd like to know what's your thoughts and if it's worth implementing it. If yes, I was only able to change the UI, but I couldn't figure out exactly how the operator to create a new project is working. I was wondering if someone could point me the right direction in the code, if it's not too advanced stuff.

**Comment by vulevukusej** [comment]
What do you expect should happen when "Number of Storeys" is 4 for example?  Should it create placeholder stories for elements to be contained in eventually?  Why would you need something like this when creating the project?

**Reply by brunoperdigao** [reply]
Yes, it would create the project with the placeholder with the proper height (and maybe [planes](https://community.osarch.org/discussion/1047/blenderbim-create-faces-for-ifcbuildingstorey)?). I actually thought of this as one of many options that one could choose before creating a project. This is useful when you already know how the project is going to be structured. I guess the best example is Freecad BIM Project Setup menu:

![image](https://user-images.githubusercontent.com/57102715/181254362-bea0d609-b458-435e-8ff5-93113f09d142.png)

![image](https://user-images.githubusercontent.com/57102715/181254635-f24ad099-d903-4a63-b534-8f87c8dfaeb9.png)

**Reply by Moult** [reply]
An optional setup wizard sounds more preferable than a single building storey field. Recently there were some QoL things implemented like when you click a building it has a button to add a storey.

![2022-08-31-183417_715x128_scrot](https://user-images.githubusercontent.com/88302/187634510-08a546f0-15de-440c-a189-091c2860642e.png)
